crates/subspace-proof-of-time/src/pot_aes.rs

```diff
@@ -0,0 +1,125 @@
+//! AES related functionality.
+
+use aes::cipher::generic_array::GenericArray;
+use aes::cipher::{BlockEncrypt, KeyInit};
+use aes::Aes128;
+use subspace_core_primitives::{PotBytes, PotCipherText, PotKey, PotSeed};
+
+pub(crate) fn create(
+    seed: PotSeed,
+    key: PotKey,
+    num_checkpoints: u64,
+    checkpoint_iterations: u64,
+) -> Vec<PotCipherText> {
+    let key = GenericArray::from(key.bytes());
+    let cipher = Aes128::new(&key);
+    let mut cur_block = GenericArray::from(seed.bytes());
+
+    let mut checkpoints = Vec::with_capacity(num_checkpoints as usize);
```

Comment:
It was actually u32  in the first cut. Changing the num_checkpoints type to usize would be the right thing to do (as it is the Vec<> size)

Reply:
I don't think so. `usize` is platform-dependent, but our protocol should not be, hence we should use exact integer sizes everywhere, but also remember about limits when converting to/from usize for any reason.

Reply:
Number of checkpoints will always be small, `u8` is enough.
Number of _iterations_ per checkpoint may be huge though.

Reply:
Let us keep it u16 just in case. iterations per checkpoint is u64 now. Should it be u128 to be safe?

Reply:
Keep iterations at u64 for now. We need to bench to see what's the actual order of magnitude of iterations

Reply:
I don't think `u16` is necessary, we'll never ever want more than 256 checkpoints.

`u128` is too excessive as well. Processors will never be able to do more than 2^64 AES iterations per second, there is literally no way that would be possible. For that processors would need to do numerous AES operations per cycle, which is not going to happen. Given the fact that we define iterations per checkpoint, I believe `checkpoint_iterations` should also become `u32` for the same reason, for 4B+ iterations you'd still need less than a cycle latency if we have one second, otherwise even less.

Reply:
Changed

Reply:
Changed to u8/u32